src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/ActionPlanControllerTest.kt

```diff
@@ -0,0 +1,83 @@
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.SampleData
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.ActionPlanService
+import java.net.URI
+import java.util.UUID
+
+internal class ActionPlanControllerTest {
+
+  private val actionPlanMapper = mock<ActionPlanMapper>()
+  private val jwtAuthUserMapper = mock<JwtAuthUserMapper>()
+  private val actionPlanService = mock<ActionPlanService>()
+  private val locationMapper = mock<LocationMapper>()
+
+  private val actionPlanController = ActionPlanController(actionPlanMapper, jwtAuthUserMapper, actionPlanService, locationMapper)
+
+  @Test
+  fun `saves draft action plan`() {
+    val referralId = UUID.randomUUID()
+    val numberOfSessions = 5
+    val activitiesDTO = emptyList<CreateActionPlanActivityDTO>()
+    val createActionPlanDTO = CreateActionPlanDTO(referralId, 5, activitiesDTO)
+    val jwtAuthenticationToken = JwtAuthenticationToken(mock())
+    val authUser = AuthUser("CRN123", "auth", "user")
+    val actionPlan = SampleData.sampleActionPlan()
+    val activities = emptyList<ActionPlanActivity>()
+    val draftActionPlanDTO = DraftActionPlanDTO.from(actionPlan)
+    val uri = URI.create("/1234")
+
+    whenever(jwtAuthUserMapper.map(jwtAuthenticationToken)).thenReturn(authUser)
+    whenever(actionPlanMapper.map(activitiesDTO)).thenReturn(activities)
+    whenever(actionPlanService.createDraftActionPlan(referralId, numberOfSessions, activities, authUser)).thenReturn(actionPlan)
+    whenever(locationMapper.map("/{id}", draftActionPlanDTO.id)).thenReturn(uri)
+
+    val draftActionPlanResponse = actionPlanController.createDraftActionPlan(createActionPlanDTO, jwtAuthenticationToken)
+
+    assertThat(draftActionPlanResponse.let { it.body }).isEqualTo(draftActionPlanDTO)
+    assertThat(draftActionPlanResponse.let { it.headers["location"] }).isEqualTo(listOf("/1234"))
+  }
+
+  @Test
+  fun `gets draft action plan using id`() {
+    val actionPlanId = UUID.randomUUID()
+    val actionPlan = SampleData.sampleActionPlan(id = actionPlanId)
+    val draftActionPlanDTO = DraftActionPlanDTO.from(actionPlan)
+
+    whenever(actionPlanService.getDraftActionPlan(actionPlanId)).thenReturn(actionPlan)
+
+    val draftActionPlanResponse = actionPlanController.getDraftActionPlan(actionPlanId.toString())
+
+    assertThat(draftActionPlanResponse).isEqualTo(draftActionPlanDTO)
+  }
+
+  @Test
+  fun `throws exception id action plan does not exist`() {
+    val actionPlanId = UUID.randomUUID()
+
+    whenever(actionPlanService.getDraftActionPlan(actionPlanId)).thenReturn(null)
+
+    val exception = assertThrows(ResponseStatusException::class.java) {
+      actionPlanController.getDraftActionPlan(actionPlanId.toString())
+    }
+
+    assertThat(exception.status).isEqualTo(NOT_FOUND)
+    assertThat(exception.reason).isEqualTo("draft action plan not found [id=$actionPlanId]")
```

Comment:
not a blocker - generally prefer not to assert on response content - this message should be changable without breaking unit tests.

Reply:
Understood